sycl/test-e2e/Matrix/joint_matrix_annotated_ptr.cpp

```diff
@@ -0,0 +1,26 @@
+//==-------- joint_matrix_annotated_ptr.cpp  - DPC++ joint_matrix-----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: matrix
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// Currently row major B fails when annotated_ptr is used
+// XFAIL: gpu
+
+#include "common.hpp"
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental::matrix;
+namespace syclintelex = sycl::ext::intel::experimental;
+namespace syclex = sycl::ext::oneapi::experimental;
```

Comment:
Let's move all such statements to .hpp to reduce duplication (I mean for all 4 .cpp files added to tests by this patch).